plugins/modules/ini_file.py

```diff
@@ -302,14 +354,22 @@ def do_ini(module, filename, section=None, option=None, values=None,
     section_lines = []
 
     for index, line in enumerate(ini_lines):
+        # end of section:
+        if within_section and line.startswith(u'['):
+            if check_section_has(
```

Comment:
This also needs to be run when the `for` loop doesn't end with a `break`, otherwise you'll miss the section if it is the very last section of the file.

(This might already be a problem with the existing code, when quickly glancing at this diff.)

Reply:
I think the sentinel value added in line 335 makes it so that the section always ends here. Let me look into adding an integration test to verify this.

Reply:
Ah right, I missed that one. Thanks for pointing this out!